### PR TITLE
fix: debug console variables output problem

### LINKED ISSUE #126967: Debug console
Issue Type: <b>Bug</b>

Can any one explain why vs code debug console work like this:

when I use console.log for logging name form array using map, at first log it work. On second log its not showing

VS Code version: Code 1.57.1 (507ce72a4466fbb27b715c3722558bb15afa9f48, 2021-06-17T13:28:07.755Z)
OS version: Windows_NT x64 10.0.19043
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 3500U with Radeon Vega Mobile Gfx   (8 x 2096)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|7.69GB (2.79GB free)|
|Process Argv|--crash-reporter-id b13aa7f5-b359-41cf-928a-e98c0833ffd2|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (9)</summary>

Extension|Author (truncated)|Version
---|---|---
insert-unicode|bru|0.10.0
bracket-pair-colorizer|Coe|1.0.61
vscode-markdownlint|Dav|0.40.4
code-runner|for|0.11.4
js-debug-nightly|ms-|2021.5.617
material-icon-theme|PKi|4.7.0
LiveServer|rit|5.6.1
vscodeintellicode|Vis|1.2.14
JavaScriptSnippets|xab|1.8.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368cf:30146710
vsreu685:30147344
python383:30185418
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
vstes627:30244334
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30311713
vspre833:30321513
pythonptprofiler:30281270
vshan820:30294714
pythondataviewer:30285071
vscus158:30321503
pythonvsuse255:30323308
vscorehov:30309549
vscod805cf:30301675
pythonvspyt200:30324779
binariesv615cf:30323120

```

</details>

### PR DESCRIPTION
**Pull request by erha19**
This PR fixes #126967

### CODE REVIEW

**Comment by isidorn** [comment]
Thank you for this PR. Unfortunately I will not have time this milestone to review this. I plan to look into this in July
Also assigning to @connor4312 as a reviewer since I think he did a PR which touches the Code your are changing.

**Review by connor4312 — CHANGES_REQUESTED** [review]
I think the idea of requesting variables before the output is dequeued is fine, but this also moves `appendToRepl` outside of the output queue which can cause logs to be displayed in incorrect order.

Also see: https://github.com/microsoft/vscode/issues/126967#issuecomment-874954269

**Comment by erha19** [comment]
thanks @connor4312 , you are right, and I have been modified my code to put `appendToRepl` into the `outputQueue`.

File: src/vs/workbench/contrib/debug/browser/debugSession.ts [hunk 1]
```diff
@@ -905,6 +905,28 @@ export class DebugSession implements IDebugSession {
...
+					source: this.getSource(event.body.source)
+				} : undefined;
+				const container = new ExpressionContainer(this, undefined, event.body.variablesReference, generateUuid());
+				await container.getChildren().then(children => {
+					children.forEach(child => {
+						// Since we can not display multiple trees in a row, we are displaying these variables one after the other (ignoring their names)
+						(<any>child).name = null;
+						// we should put appendToRepl into queue to make sure the logs to be displayed in correct order
+						// see https://github.com/microsoft/vscode/issues/126967#issuecomment-874954269
+						outputQueue.queue(async () => {
```
Comment (connor4312) [inline src/vs/workbench/contrib/debug/browser/debugSession.ts:922]: The enqueuing of output must happen synchronously to preserve order. The `onDidOutput` listener should not be async at all.

What you can do is something like

```
const children = container.getChildren();
outputQueue.queue(async () => {
    const resolved = await children;
    resolved.forEach(() => ...)
});
```

Comment (erha19) [reply src/vs/workbench/contrib/debug/browser/debugSession.ts]: done.

**Review by connor4312 — APPROVED** [review]

**Comment by erha19** [comment]
Hi @isidorn, the PR can be merged.

**Comment by erha19** [comment]
@connor4312 hi, do you know who can review or merge this PR？

**Comment by isidorn** [comment]
Thanks for this PR! This week we are in the endgame which means we are not accepting any new code changes.
So I am assigning this to August so we potentially merge this start of next week.
Thanks!

**Comment by isidorn** [comment]
As @connor4312 mentioned this PR is an improvement so let's merge it in.
@erha19 thanks a lot for your contribution 👏

**Pull request merged**